src/lsp/cobol_indent/cobol_indent.ml

```diff
@@ -16,10 +16,62 @@ module Type = Indent_type
 (*return the result of indentation. use user-defined indent_config*)
 let indent_range = Indenter.indent_range
 
-(*indent the whole file and print*)
-let indent_file ~dialect ~source_format ~file ~indent_config =
-  let contents = Ez_file.V1.EzFile.read_file file in
-  indent_range
-    ~dialect ~source_format ~range:None ~indent_config
-    ~filename:file ~contents
-  |> Fmt.pr "%s"
\ No newline at end of file
+(** [apply s rdl] applies the changes in [rdl] to the text in [s]. [rdl] is
+    expected to have been computed by calling [indent_range ~contents:s]. *)
+let apply =
+  (*
+    [irs] contains a list of {!Type.indent_record} entries describing how to
+     modify the indentation.  They must be sorted in increasing order, and there
+     must be at most one record per line.
+
+    [buf] is a {!Buffer.t} into which we are writing the indented content.
+
+    [s] is the original content to indent as a {!string}.
+
+    [line] is the current line, which starts at offset [bol] in the string [s].
+
+    [start] is an offset in [s], less than or equal to [bol], that must be
+    preserved as-is. This allows copying large unmodified chunks of text (lines
+    that are already properly indented) using a single call to
+    {!Buffer.add_substring} for reduced overhead.
+  *)
+  let rec aux ~start (irs : Type.indent_record list) buf s line bol =
+    (* Note: this can raise [Invalid_argument] if the modifications do not
+       cleanly apply to the input string, which include cases where there are
+       modifications on lines past the end of the input, modifications that are
+       not sorted, etc. -- we can call [invalid_arg] ourselves, but
+       [String.index_from] can also raise [Invalid_argument] in these
+       conditions. *)
+    match irs with
+    | [] ->
+      if start = 0 then s else (
+        Buffer.add_substring buf s start (String.length s - start);
+        Buffer.contents buf
+      )
+    | { lnum; offset_orig; offset_modif } :: irs when line = lnum - 1 ->
+      let eol =
+        try String.index_from s bol '\n' + 1 with Not_found -> String.length s
+      in
+      let delta = offset_modif - offset_orig in
+      if delta > 0 || true then (
```

Comment:
Why `|| true` (which kills `else` branch)?

Reply:
Because this is a leftover from some debugging, good catch (note that this code is only used in the `superbol indent` command).